tools/torch_quant/torch_quant/quantizer.py

```diff
@@ -258,26 +234,48 @@ def qat_gm(
             w_ob_ctr=ob_types.w_ob_ctr,
             bias_ob_ctr=ob_types.bias_ob_ctr,
         )
-        ctx.modify_graph([
-            set_qconfig,
-            insert_act_observer,
-            # Generally we do not add fake-quant to bias during qat fine-tuning. If
-            # users want to evaluate the accuracy of a model in specific state, they
-            # should use the model returned by `quantizer.quantize`
-            partial(quantizable_module_to_observed, is_observe_bias=False)])
+        if self.backend == Backend.DISC:
+            ctx.modify_graph([
+                set_qconfig,
+                insert_act_observer,
+                insert_w_observer,
+                # Generally we do not add fake-quant to bias during qat fine-tuning. If
+                # users want to evaluate the accuracy of a model in specific state, they
+                # should use the model returned by `quantizer.quantize`
+                quantizable_module_to_observed])
+        else:
+            ctx.act_ob_ctr = self.act_ob_ctr
+            ctx.w_ob_ctr = self.w_ob_ctr
+            ctx.bias_ob_ctr = self.bias_ob_ctr
+            ctx.modify_graph([set_qconfig, fuse_modules, insert_w_observer])
+
+            ctx.act_ob_ctr = ob_types.act_ob_ctr
+            ctx.w_ob_ctr = ob_types.w_ob_ctr
+            ctx.bias_ob_ctr = ob_types.bias_ob_ctr
+            ctx.modify_graph(
+                [
+                    insert_act_observer,
+                    insert_w_observer,
```

Comment:
What is the purpose of insert weight observer alone instead of the origin one.

Reply:
In early version, we only inserted weight observers for DISC backend, which was implemented in `quantizable_module_to_observed` pass. But now it is also needed by other backends for performing amp or qat.